soletta/simple_type_c.go

```diff
@@ -0,0 +1,466 @@
+package soletta
+
+/*
+#include <ctype.h>
+#include <stdlib.h>
+#include <stdarg.h>
+#include <sol-log.h>
+#include <sol-flow.h>
+
+#include "simple_type_c.h"
+
+#define streq(a, b) (strcmp((a), (b)) == 0)
+
+struct sol_flow_node {
+    const struct sol_flow_node_type *type;
+    struct sol_flow_node *parent;
+    char *id;
+
+    void *parent_data;
+
+    void *data[];
+};
```

Comment:
I copied the definition here because the field type is accessed at line 163:  "const struct simple_c_type_type_data *type_data = node->type->type_data;". This could be avoided if there were some kind of node type data access function.